samples/LibVLCSharp.CustomRendering.Direct3D11/Program.cs

```diff
@@ -0,0 +1,682 @@
+
+            _d3dctx->VSSetShader(pVS, null, 0);
+            _d3dctx->PSSetShader(pPS, null, 0);
+
+            var samplerDesc = new D3D11_SAMPLER_DESC
+            {
+                Filter = D3D11_FILTER_MIN_MAG_LINEAR_MIP_POINT,
+                AddressU = D3D11_TEXTURE_ADDRESS_CLAMP,
+                AddressV = D3D11_TEXTURE_ADDRESS_CLAMP,
+                AddressW = D3D11_TEXTURE_ADDRESS_CLAMP,
+                ComparisonFunc = D3D11_COMPARISON_ALWAYS,
+                MinLOD = 0,
+                MaxLOD = D3D11_FLOAT32_MAX
+            };
+
+            fixed (ID3D11SamplerState** ss = &samplerState)
+            {
+                ThrowIfFailed(_d3dDevice->CreateSamplerState(&samplerDesc, ss));
+                _d3dctx->PSSetSamplers(0, 1, ss);
+            }
+
+            fixed (RTL_CRITICAL_SECTION* sl = &sizeLock)
+                InitializeCriticalSection(sl);
+        }
+
+        static void InitializeLibVLC()
+        {
+            Core.Initialize();
+
+            libvlc = new LibVLC(enableDebugLogs: true);
+            libvlc.Log += (s, e) => Debug.WriteLine(e.FormattedLog);
+
+            mediaplayer = new MediaPlayer(libvlc);
+
+            using var media = new Media(libvlc, new Uri("http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4"));
+            mediaplayer.Media = media;
+
+            mediaplayer.SetOutputCallbacks(VideoEngine.D3D11, OutputSetup, OutputCleanup, OutputSetResize, UpdateOuput, Swap, StartRendering, null, null, SelectPlane);
+
+            mediaplayer.Play();
+        }
+
+        static ID3D11Buffer* CreateBuffer(D3D11_BUFFER_DESC bd)
+        {
+            ID3D11Buffer* buffer;
+
+            ThrowIfFailed(_d3dDevice->CreateBuffer(&bd, null, &buffer));
+
+            return buffer;
+        }
+
+        static readonly string shaderStr = @"
+Texture2D shaderTexture;
+SamplerState samplerState;
+struct PS_INPUT
+{
+    float4 position     : SV_POSITION;
+    float4 textureCoord : TEXCOORD0;
+};
+
+float4 PShader(PS_INPUT In) : SV_TARGET
+{
+    return shaderTexture.Sample(samplerState, In.textureCoord);
+}
+
+struct VS_INPUT
+{
+    float4 position     : POSITION;
+    float4 textureCoord : TEXCOORD0;
+};
+
+struct VS_OUTPUT
+{
+    float4 position     : SV_POSITION;
+    float4 textureCoord : TEXCOORD0;
+};
+
+VS_OUTPUT VShader(VS_INPUT In)
+{
+    return In;
+}
+";
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct ShaderInput
+        {
+            internal Position position;
+            internal Texture texture;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct Position
+        {
+            internal float x;
+            internal float y;
+            internal float z;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct Texture
+        {
+            internal float x;
+            internal float y;
```

Comment:
Should we ship these structs as part of LibVLCSharp? They'll likely be needed for all code implementing this API (at least the default use case)

Reply:
Probably not. It's used in the sample app but not in other implementations (like vlc-unity). Let users implement their own structs if needed.